File: BAEKJOON/Python/1000/1000/1032.py
"""
Title : 명령 프롬프트
Link  : https://www.acmicpc.net/problem/1032
"""

from sys import stdin

input = stdin.readline


if __name__ == "__main__":
    N = int(input())
    commands = list(input().strip() for _ in range(N))
    cmd_length = len(commands[0])
    ans = []
    for j in range(cmd_length):
        char = commands[0][j]
        for i in range(1, N):
            if commands[i][j] != char:
                ans.append("?")
                break
        else:
            ans.append(char)
    print("".join(ans))

File: BAEKJOON/Javascript/1000/1000/1032.js
const fs = require("fs")
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt"
const input = fs.readFileSync(filePath).toString().trim().split("\n")


const N = parseInt(input.shift())

let ans = []
const cmdLength = input[0].length
for (j = 0; j < cmdLength; j++) {
  let char = input[0][j]
  let flag = true
  for (i = 0; i < N; i++) {
    if (input[i][j] !== char) {
      flag = false
      i = N
    }
  }

  if (flag === true) {
    ans.push(char)
  } else {
    ans.push("?")
  }
}
console.log(ans.join(""))
